delay/delay_pre17.go

```diff
@@ -0,0 +1,19 @@
+// Copyright 2017 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+//+build !go1.7
+
+package delay
+
+import (
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var contextType = reflect.TypeOf((*context.Context)(nil)).Elem()
+
+func isContext(t reflect.Type) bool {
```

Comment:
How about using `Type.Implements`?

https://golang.org/pkg/reflect/#Type

Reply:
`Type.Implements` would be wrong since it would permit a concrete type that implements `Context` to be the first argument.

`Type.AssignableTo` is interesting... that could work, although likely we'd want to have any interface that has exactly the same method set, since `AssignableTo` could accept an interface that's smaller than Context.